quinn-h3/src/body.rs

```diff
@@ -107,29 +108,44 @@ pub struct RecvBody {
 }
 
 impl RecvBody {
-    pub(crate) fn with_capacity(
+    pub(crate) fn new(
         recv: FrameStream,
-        capacity: usize,
-        max_size: usize,
         conn: ConnectionRef,
         stream_id: StreamId,
     ) -> Self {
-        if capacity < 1 {
-            panic!("capacity cannot be 0");
-        }
-
         Self {
-            max_size,
             conn,
             stream_id,
-            state: RecvBodyState::Receiving(BytesMut::with_capacity(capacity)),
-            body: None,
             recv,
+            body: None,
+            max_size: RECV_BODY_MAX_SIZE_DEFAULT,
+            capacity: RECV_BODY_CAPACITY_DEFAULT,
+            state: RecvBodyState::Initial,
         }
     }
+
+    pub fn with_capacity(
+        mut self,
+        capacity: usize,
+        max_size: usize,
+    ) -> Self {
+        match &self.state {
```

Comment:
Since we don't need the actual value of `self.state` here, this would be more idiomatic (and slightly more compact) as an `if let RecvBodyState::Initial =`  block.

Reply:
This has been removed in a later commit.

Reply:
Ah, never mind then!